src/crystal/compiler_rt/udivmodti4.cr

```diff
@@ -0,0 +1,127 @@
+# Function return the remainder of the unsigned division with overflow eg. `a % b`
+
+fun __udivmodti4(a : UInt128, b : UInt128, rem : UInt128*) : UInt128
+  n_udword_bits = sizeof(Int64) &* sizeof(Char)
+  n_utword_bits = sizeof(Int128) &* sizeof(Char)
+  n = a.unsafe_as(CompilerRT::UI128)
+  d = b.unsafe_as(CompilerRT::UI128)
+  q = CompilerRT::UI128.new
+  r = CompilerRT::UI128.new
+  sr = 0_u32
+
+  if n.info.high == 0
+    if (d.info.high == 0)
+      if rem
+        rem.value = (n.info.low % d.info.low).to_u128
+      end
+      n.info.low = n.info.low // d.info.low
+      return n.all
+    end
+    rem.value = n.info.low.to_u128 if rem
+    return 0_u128
```

Comment:
Is this intentional?

Reply:
this is intentional it is ported from here https://github.com/llvm-mirror/compiler-rt/blob/master/lib/builtins/udivmodti4.c#L47

Reply:
Making the code below essentially pointless...? I don't get it :/